http_2.rflx

```diff
@@ -0,0 +1,230 @@
+			Frame_Type : Frame_Type;
+			Flag_End_Stream_ACK : Boolean
+				then Bit_2_Unused
+					if ((Frame_Type = SETTINGS and Payload_Length = 6) or
+						(Frame_Type = PING and Payload_Length = 8) or
+						(Frame_Type = DATA) or
+						(Frame_Type = HEADERS) or
+						((	Frame_Type = PUSH_PROMISE or
+							Frame_Type = CONTINUATION or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							(Frame_Type = WINDOW_UPDATE and Payload_Length = 4) or
+							(Frame_Type = PRIORITY and Payload_Length = 5)
+						) and Flag_End_Stream_ACK = False));
+			Bit_2_Unused : Zero_Bit;
+			Flag_End_Headers : Boolean
+				then Flag_Padded
+					if (Frame_Type = HEADERS or
+						Frame_Type = PUSH_PROMISE or
+						Frame_Type = CONTINUATION or
+						((	Frame_Type = DATA or
+							Frame_Type = SETTINGS or
+							Frame_Type = PING or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							Frame_Type = WINDOW_UPDATE or
+							Frame_Type = PRIORITY
+						) and Flag_End_Headers = False));
+			Flag_Padded : Boolean
+				then Bit_5_Unused
+					if (Frame_Type = DATA or
+						Frame_Type = HEADERS or
+						Frame_Type = PUSH_PROMISE or
+						((	Frame_Type = SETTINGS or
+							Frame_Type = PING or
+							Frame_Type = CONTINUATION or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							Frame_Type = WINDOW_UPDATE or
+							Frame_Type = PRIORITY
+						) and Flag_Padded = False));
+			Bit_5_Unused : Zero_Bit;
+			Flag_Priority : Boolean
+				then Bit_7_Unused
+					if (Frame_Type = HEADERS or
+						((	Frame_Type = DATA or
+							Frame_Type = SETTINGS or
+							Frame_Type = PING or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							Frame_Type = WINDOW_UPDATE or
+							Frame_Type = PRIORITY or
+							Frame_Type = PUSH_PROMISE or
+							Frame_Type = CONTINUATION
+						) and Flag_Priority = False));
+			Bit_7_Unused : Zero_Bit;
+			Bit_8_Unused : Zero_Bit;
+			Reserved_Bit_1 : Zero_Bit;
+			Stream_Identifier : Stream_ID 	-- Last Header Field
+				then Pad_Length
+					if (Frame_Type = DATA or Frame_Type = HEADERS or Frame_Type = PUSH_PROMISE) and Flag_Padded = True
+				then Data
+					with Size => Payload_Length * 8
+					if Frame_Type = DATA and Flag_Padded = False
+				then Exclusive_Flag
+					if Frame_Type = HEADERS and Flag_Priority = True and Flag_Padded = False
+				then Header_Block_Fragment
+					with Size => Payload_Length * 8
+					if (Frame_Type = HEADERS and Flag_Priority = False and Flag_Padded = False) or Frame_Type = CONTINUATION
+				then Exclusive_Flag
+					if Frame_Type = PRIORITY
+				then Error_Code
+					if Frame_Type = RST_STREAM
+				then Settings_Identifier
+					if Frame_Type = SETTINGS
+				then Data
+					with Size => 64
+					if Frame_Type = PING and Stream_Identifier = 0 -- RFC: If a PING frame is received with a stream identifier field value other than 0x0, the recipient MUST respond with a connection error (Section 5.4.1) of type PROTOCOL_ERROR.
```

Comment:
The comment describes behavior which should be modeled in the session specification. It is not helpful here.

Reply:
Removed in 31d9594